src/influxdb_ioxd/rpc/storage/data.rs

```diff
@@ -356,6 +358,111 @@ fn datatype_to_measurement_field_enum(data_type: &ArrowDataType) -> Result<Field
     }
 }
 
+/// Wrapper struture that implements [`std::fmt::Display`] for a slice
+/// of `Frame`s
+struct DisplayableFrames<'a> {
+    frames: &'a [Frame],
+}
+
+impl<'a> DisplayableFrames<'a> {
+    fn new(frames: &'a [Frame]) -> Self {
+        Self { frames }
+    }
+}
+
+impl<'a> fmt::Display for DisplayableFrames<'a> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        for frame in self.frames.iter() {
+            format_frame(frame, f)?;
+            writeln!(f)?
+        }
+        Ok(())
+    }
```

Comment:
For loop is fine here, but when I see a `for` and an `Ok(())` it does make me think it's one of the few times that `try_for_each` is suitable:

```suggestion
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        self.frames.iter().try_for_each(|frame| {
            format_frame(frame, f)?;
            writeln!(f)
        })
    }
```